### PR TITLE
fix: schema with `properties` and `oneOf` properties

### PR DESCRIPTION
**Pull request by gr2m**
follow up to https://github.com/octokit/types.ts/issues/274 and https://github.com/github/rest-api-description/issues/220

### CODE REVIEW

**Comment by drwpow** [comment]
@gr2m This looks like it’s WIP? Happy to review whenever; just tag me!

**Comment by gr2m** [comment]
Yea it’s WIP, just added a failing test so far. I’ll let you know ones it’s ready

**Comment by gr2m** [comment]
@drwpow this is ready for review. In the first two commits I only update the OpenAPI spec from GitHub and our generated `github.ts` to bump the test baseline to latest. The 3rd commit is the tests I introduced, the 4th commit is the code changes. And finally [the 5th commit](https://github.com/drwpow/openapi-typescript/pull/491/commits/14b7de8f59ead8111f62ced33a9d8c8744a54f5a) shows that this change fixes the problem that came up in https://github.com/octokit/types.ts/issues/274

**Comment by drwpow** [comment]
> I only update the OpenAPI spec from GitHub

That works for me! I’d like for the `specs/*` folders to be real-world schemas we’re testing against. In the past, I’ve only been against contributors adding spec test-like additions to `specs/*`, because the whole point is to update these as the specs evolve, and we don’t want their changes to be blown away. But I’d love to keep `specs/*` updated so we always ensure we’re generating types correctly.

File: src/transform/schema.ts [hunk 1]
```diff
@@ -105,9 +111,12 @@ export function transformSchemaObj(node: any): string {
 
       output += tsIntersectionOf([
         ...(node.allOf ? (node.allOf as any[]).map(transformSchemaObj) : []), // append allOf first
+        ...(node.anyOf ? [transformAnyOf(node.anyOf)] : []), // append allOf first
+        ...(node.oneOf ? [transformOneOf(node.oneOf)] : []), // append allOf first
```
Comment (drwpow) [inline src/transform/schema.ts:115]: 👍🏻

File: src/transform/schema.ts [hunk 2]
```diff
@@ -77,8 +77,14 @@ export function transformSchemaObj(node: any): string {
       break;
     }
     case "object": {
+      const isAnyOfOrOneOfOrAllOf = "anyOf" in node || "oneOf" in node || "allOf" in node;
```
Comment (drwpow) [inline src/transform/schema.ts:80]: 😄 this logic has always broken my brain. This is clearer! Thanks

File: tests/schema.test.ts [hunk 3]
```diff
@@ -213,6 +213,57 @@ describe("SchemaObject", () => {
...
+      ).toBe(`((Partial<{
+"b": string;
+
+}>) & (Partial<{
+"c": string;
+
+}>)) & ({
+"a"?: string;
+
+})`);
```
Comment (drwpow) [inline tests/schema.test.ts:265]: Nice test 👍🏻

**Review by drwpow — APPROVED** [review]

File: src/transform/schema.ts (hunk 1 above)
Comment (gr2m) [inline src/transform/schema.ts:115]:
```suggestion
        ...(node.anyOf ? [transformAnyOf(node.anyOf)] : []), // append anyOf first
        ...(node.oneOf ? [transformOneOf(node.oneOf)] : []), // append oneOf first
```

File: src/transform/schema.ts [hunk 4]
```diff
@@ -105,9 +111,12 @@ export function transformSchemaObj(node: any): string {
 
       output += tsIntersectionOf([
         ...(node.allOf ? (node.allOf as any[]).map(transformSchemaObj) : []), // append allOf first
+        ...(node.anyOf ? [transformAnyOf(node.anyOf)] : []), // append anyOf first
+        ...(node.oneOf ? [transformOneOf(node.oneOf)] : []), // append oneOf first
```
Comment (gr2m) [inline src/transform/schema.ts:115]:
```suggestion
        // append allOf/anyOf/oneOf first
        ...(node.allOf ? (node.allOf as any[]).map(transformSchemaObj) : []),
        ...(node.anyOf ? [transformAnyOf(node.anyOf)] : []),
        ...(node.oneOf ? [transformOneOf(node.oneOf)] : []),
```

**Comment by gr2m** [comment]
@drwpow  could you cut a new release please?

**Comment by drwpow** [comment]
Sure!

**Comment by drwpow** [comment]
`3.0.3` released

**Pull request merged**